### PR TITLE
JFF-3 Create Initiate Issuance request object

### PR DESCRIPTION
**Pull request by maikel-maas**
- Created the IssuanceInitiationRequest interface;
   - Created function to parse URI into json;
   - Created function to parse json into URI;
   - Created enum for the authorization flows;
   - Created valueOf function with request object param;
   - Wrote basic README.md file;
   - Added unit tests;

### CODE REVIEW

File: src/functions/Encoding.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  function encodeAndStripWhitespace(key: string) {
+    return encodeURIComponent(key.replace(' ', ''));
+  }
+
+  for (const [key, value] of Object.entries(json)) {
+    if (!value) {
+      continue;
+    }
+    //Skip properties that are not of URL type
+    if (!['issuer', 'credential_type'].includes(key)) {
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply src/functions/Encoding.ts]: Fixed!

File: src/functions/Encoding.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  }
+  return results.join('&');
+}
+
+/**
+ * @function decodeUriAsJson decodes an URI into a Json object
+ * @param uri string
+ */
+export function decodeURIAsJson(uri: string): IssuanceInitiationRequestPayload[] | IssuanceInitiationRequestPayload {
+  if (!uri || !uri.includes('issuer') || !uri.includes('credential_type')) {
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply src/functions/Encoding.ts]: Fixed!

File: __tests__/IssuanceInitiationRequest.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      issuer: 'test',
+      credential_type: 'test'
+    })).toEqual(AuthzFlowType.AUTHORIZATION_CODE_FLOW)
+  })
+  it('should return pre-authorized code flow', () => {
+    expect(AuthzFlowType.valueOf({
+      issuer: 'test',
+      credential_type: 'test',
+      pre_authorized_code: 'test'
+    })).toEqual(AuthzFlowType.PRE_AUTHORIZED_CODE_FLOW)
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply __tests__/IssuanceInitiationRequest.test.ts:38]: Fixed!

File: __tests__/IssuanceInitiationRequest.test.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    expect(encodeJsonAsURI({
+      issuer: 'https://server.example.com',
+      credential_type: 'https://did.example.org/healthCard',
+      op_state: 'eyJhbGciOiJSU0Et...FYUaBy'
+    })).toEqual('issuer=https%3A%2F%2Fserver%2Eexample%2Ecom&credential_type=https%3A%2F%2Fdid%2Eexample%2Eorg%2FhealthCard&op_state=eyJhbGciOiJSU0Et...FYUaBy')
+  })
+  it('should parse an array of objects as open-id-URI', () => {
+    expect(encodeJsonAsURI([
+    {
+      issuer: 'https://server.example.com',
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply __tests__/IssuanceInitiationRequest.test.ts]: Fixed!

File: README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+//         op_state: 'eyJhbGciOiJSU0Et...FYUaBy'
+//     },
+//     {
+//       issuer: 'https://server.example1.com',
+//           credential_type: 'https://did.example1.org/healthCard',
+//         op_state: 'eyJhbGciOiJSU0Et...FYUaBy'
+//     },
+// ]
+```
+* NOTE: The input may contain duplicated keys, that will result in an array
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply README.md]: Fixed!

File: README.md [hunk 6]
```diff
@@ -8,16 +8,40 @@
...
+The below diagram shows the steps involved in this flow. Note that wallet inner functionalities (like saving VCs) are out of scope of this library. Also This library doesn't involve any functionalities of a VC Issuer
+![Flow diagram](https://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/Sphereon-Opensource/OIDC4VCI-client/develop/docs/preauthorized-code-flow.puml)
+
 ### Interfaces
 
 ```typescript
 export interface IssuanceInitiationRequestPayload {
   issuer: string, //REQUIRED The issuer URL of the Credential issuer, the Wallet is requested to obtain one or more Credentials from.
-  credential_type: string[] | string, //REQUIRED A JSON string denoting the type of the Credential the Wallet shall request
+  credential_type: string[], //REQUIRED A JSON string denoting the type of the Credential the Wallet shall request
```
Comment (nklomp) [inline README.md:23]: Why is the singular string removed? The actual example in the spec uses a singular string for it

**Review by nklomp — CHANGES_REQUESTED** [review]
credential_type is either a singular string or a string array

File: README.md (hunk 6 above)
Comment (maikel-maas) [reply README.md:23]: Fixed

**Review by nklomp — APPROVED** [review]

**Pull request merged**